### PR TITLE
openvino: fix aarch64 build

### PR DESCRIPTION
**Pull request by leonm1**
This package was previously marked broken on aarch64. Adding scons as a native build input, blocking scons's build hooks, and removing x86-specific build options (for intel processors) seems to have fixed the build.

I noticed a build error when not using the cmake build dir[1], so I also removed that option.

* [1] cmake build dir error:
```
[  6%] Linking CXX static library ../../../../bin/aarch64/Release/libprotobuf.a
cd /build/source/thirdparty/protobuf/protobuf/cmake && /nix/store/wjd2vl5fj4ljip18xc8yi0n0c92rw57g-cmake-3.27.9/bin/cmake -P CMakeFiles/libprotobuf.dir/cmake_clean_target.cmake
cd /build/source/thirdparty/protobuf/protobuf/cmake && /nix/store/wjd2vl5fj4ljip18xc8yi0n0c92rw57g-cmake-3.27.9/bin/cmake -E cmake_link_script CMakeFiles/libprotobuf.dir/link.txt --ve>
/nix/store/ys1miqb0vwsfkrmhv1c499wd8x8mzai2-binutils-wrapper-2.40/bin/ar qc ../../../../bin/aarch64/Release/libprotobuf.a CMakeFiles/libprotobuf.dir/__/src/google/protobuf/any_lite.cc>
/nix/store/ys1miqb0vwsfkrmhv1c499wd8x8mzai2-binutils-wrapper-2.40/bin/ranlib ../../../../bin/aarch64/Release/libprotobuf.a
make[2]: Leaving directory '/build/source'
[  6%] Built target libprotobuf
make[1]: Leaving directory '/build/source'
make: *** [Makefile:159: all] Error 2
```
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/openvino/default.nix [hunk 1]
```diff
@@ -125,8 +129,8 @@ stdenv.mkDerivation rec {
     (cmakeBool "ENABLE_SAMPLES" false)
 
     # features
-    (cmakeBool "ENABLE_INTEL_CPU" true)
-    (cmakeBool "ENABLE_INTEL_GNA" true)
+    (cmakeBool "ENABLE_INTEL_CPU" (! stdenv.isAarch64))
+    (cmakeBool "ENABLE_INTEL_GNA" (! stdenv.isAarch64))
```
Comment (jopejoe1) [inline pkgs/development/libraries/openvino/default.nix:133]: This probly also doesn't apply to other architectures so it probaly should only be trie when isx86 maybe even only on isx86_64 depending if this supports 32 bit or not

```suggestion
    (cmakeBool "ENABLE_INTEL_CPU" stdenv.isx86)
    (cmakeBool "ENABLE_INTEL_GNA" stdenv.isx86)
```

Comment (leonm1) [reply pkgs/development/libraries/openvino/default.nix]: Done. Thanks for the suggestion!

**Comment by mweinelt** [comment]
Needs a rebase, and please get rid of that merge commit.

**Comment by mweinelt** [comment]
Picking this into the openvino 2024.2.0 update PR.

**Pull request closed without merging**